File: lessons/00-workflow-or-agent/python/src/boundrelay_m0/deterministic_router.py
from .types import RouteDecision

_BILLING_KEYWORDS = ("charged", "charge", "invoice", "payment", "refund", "billed")
_TECHNICAL_KEYWORDS = ("error", "crash", "cannot log in", "can't log in", "bug", "broken")


def classify_deterministically(request: str) -> RouteDecision:
    value = request.lower()
    if any(keyword in value for keyword in _BILLING_KEYWORDS):
        return RouteDecision("billing", 1.0)
    if any(keyword in value for keyword in _TECHNICAL_KEYWORDS):
        return RouteDecision("technical", 1.0)
    return RouteDecision("general", 1.0)

File: lessons/00-workflow-or-agent/typescript/src/deterministic-router.ts
import type {RouteDecision} from "./types.js";

const BILLING_KEYWORDS = ["charged", "charge", "invoice", "payment", "refund", "billed"] as const;
const TECHNICAL_KEYWORDS = ["error", "crash", "cannot log in", "can't log in", "bug", "broken"] as const;

export function classifyDeterministically(request: string): RouteDecision {
  const value = request.toLowerCase();
  if (BILLING_KEYWORDS.some((keyword) => value.includes(keyword))) {
    return {route: "billing", confidence: 1};
  }
  if (TECHNICAL_KEYWORDS.some((keyword) => value.includes(keyword))) {
    return {route: "technical", confidence: 1};
  }
  return {route: "general", confidence: 1};
}
